presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/TableMetadataSystemTable.java

```diff
@@ -13,40 +13,68 @@
  */
 package com.facebook.presto.raptor.systemtables;
 
+import com.facebook.presto.raptor.metadata.ColumnMetadataRow;
 import com.facebook.presto.raptor.metadata.ForMetadata;
+import com.facebook.presto.raptor.metadata.MetadataDao;
+import com.facebook.presto.raptor.metadata.TableMetadataRow;
 import com.facebook.presto.spi.ColumnMetadata;
 import com.facebook.presto.spi.ConnectorPageSource;
 import com.facebook.presto.spi.ConnectorSession;
 import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.FixedPageSource;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.PrestoException;
 import com.facebook.presto.spi.SchemaTableName;
 import com.facebook.presto.spi.SystemTable;
+import com.facebook.presto.spi.block.BlockBuilder;
 import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.predicate.Domain;
 import com.facebook.presto.spi.predicate.TupleDomain;
 import com.facebook.presto.spi.type.Type;
 import com.facebook.presto.spi.type.TypeManager;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.PeekingIterator;
+import io.airlift.slice.Slice;
 import org.skife.jdbi.v2.IDBI;
 
 import javax.inject.Inject;
 
-import static com.facebook.presto.raptor.systemtables.TableMetadataPageSource.SCHEMA_NAME;
-import static com.facebook.presto.raptor.systemtables.TableMetadataPageSource.TABLE_NAME;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import static com.facebook.presto.raptor.RaptorErrorCode.RAPTOR_CORRUPT_METADATA;
+import static com.facebook.presto.raptor.util.DatabaseUtil.onDemandDao;
+import static com.facebook.presto.raptor.util.Types.checkType;
 import static com.facebook.presto.spi.SystemTable.Distribution.SINGLE_COORDINATOR;
 import static com.facebook.presto.spi.type.BigintType.BIGINT;
 import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
 import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.collect.Iterators.peekingIterator;
+import static io.airlift.slice.Slices.utf8Slice;
+import static java.lang.String.format;
+import static java.util.Locale.ENGLISH;
 import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
 
 public class TableMetadataSystemTable
         implements SystemTable
 {
-    private final IDBI dbi;
+    private static final String TABLE_NAME = "table_name";
+    private static final String SCHEMA_NAME = "table_schema";
+
+    private final MetadataDao dao;
     private final ConnectorTableMetadata tableMetadata;
 
     @Inject
     public TableMetadataSystemTable(@ForMetadata IDBI dbi, TypeManager typeManager)
     {
-        this.dbi = requireNonNull(dbi, "dbi is null");
```

Comment:
Good catch, I'll move this check to `onDemandDao` which should have one since it is capturing the `IDBI` instance for later use.